spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -102,6 +101,7 @@
     this.replacePartitions = replacePartitions;
     this.applicationId = applicationId;
     this.wapId = wapId;
+    this.targetFileSize = options.getLong("target-file-size", Long.MAX_VALUE);
```

Comment:
Do you intend to make this a table property as well? I think it makes sense as table configuration.

Reply:
+1 on having this as a table property. It can be used by compaction jobs as well.

Reply:
Good point about compaction.

Reply:
Makes sense. Will add.